### PR TITLE
Client info page and component

### PR DESCRIPTION
**Pull request by Andree37**
This needs to be integrated with the calendar component @tiagoneto7 is creating and the future payment component.

### CODE REVIEW

File: src/components/Forms/RegisterForm.tsx [hunk 1]
```diff
@@ -14,18 +14,18 @@ async function doRegister(
...
     });
 
     // this cookie part is specific to this register request
     // mustn't appear on other calls
-    setCookie('jwt-cookie', jwt, {
+    setCookie('jwt-cookie', response?.jwt, {
         path: '/',
         maxAge: 60 * 60 * 24 * 7,
-        sameSite: 'lax',
+        sameSite: 'strict',
```
Comment (tommasini) [inline src/components/Forms/RegisterForm.tsx:28]: Can you explan the diff here?

File: src/components/Forms/UserInfoForm.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
+'use client';
```
Comment (tommasini) [inline src/components/Forms/UserInfoForm.tsx:1]: On this component, it seems okay, but I'm afraid if we have some with more logic, we shouldn't adopt a different structure folder like, creating a folder for UseInfoForm and inside of it we could have an util folder with the util functions. Let me know what is your thoughts

File: src/lib/fetchers/clientFetcher.ts [hunk 3]
```diff
@@ -11,7 +10,7 @@ type FetcherParameters = {
 };
 
 // POST, PUT, DELETE fetcher
-export function clientFetcher(params: FetcherParameters) {
+export function clientFetcher<DataType>(params: FetcherParameters): Promise<DataType | undefined> {
```
Comment (tommasini) [inline src/lib/fetchers/clientFetcher.ts]: Shouldn't we get rid of the undefined like we talked?

**Review by tommasini — CHANGES_REQUESTED** [review]
Address comments my friend, but it's beautifull

File: src/components/Forms/RegisterForm.tsx (hunk 1 above)
Comment (Andree37) [reply src/components/Forms/RegisterForm.tsx:28]: Security reasons, mostly :D 

> Lax allows the cookie to be sent on some cross-site requests, whereas Strict never allows the cookie to be sent on a cross-site request.

https://stackoverflow.com/questions/59990864/what-is-the-difference-between-samesite-lax-and-samesite-strict

File: src/components/Forms/UserInfoForm.tsx (hunk 2 above)
Comment (Andree37) [reply src/components/Forms/UserInfoForm.tsx:1]: Do you mean with the amount of functions?

It depends on the way you want to approach this. Following other languages like Rust or even old ways of React, a lot of the logic that is intrinsic with the component is within itself.
I do think that repeating this code is less hurtful than trying to make it fit for all possible cases and eventually becomes dead code.

We can discuss this though :)

File: src/lib/fetchers/clientFetcher.ts (hunk 3 above)
Comment (Andree37) [reply src/lib/fetchers/clientFetcher.ts]: Done on new revision

File: src/components/Forms/UserInfoForm.tsx (hunk 2 above)
Comment (tommasini) [reply src/components/Forms/UserInfoForm.tsx:1]: Just wondering function that are used across the platform like some util functions, and can be reused on a future feature, it's time that we save I think

**Review by tommasini — APPROVED** [review]
LGTM

**Pull request merged**